### ISSUE TITLE
Debugger doesn't start in a specific Conda Environment

### ISSUE THREAD

**Issue by ttopac** [microsoft/vscode-jupyter#12556]
Type: <b>Bug</b>

# Behaviour
## Expected vs. Actual

I expect debugger to start, but it doesn't start when I'm in the conda environment named "python39_tf". It starts when the code is run in other environments.

## Steps to reproduce:

Alternative A:
1a. Click on the "debug" button in the "Run and Debug" pane.
2a. Loading bar goes halfway, then nothing happens.

Alternative B:
1b. Click the debug cell in Jupyter.
2b. The Jupyter cell gives a runtime error and provides the following (hard to follow) Jupyter log.

# Diagnostic data

-   Python version (& distribution if applicable, e.g. Anaconda): 3.9.15
-   Type of virtual environment used (e.g. conda, venv, virtualenv, etc.): Conda
-   Value of the `python.languageServer` setting: Default

<details>

<summary>
Output for <code>Python</code> doesn't provide any useful information, but the Jupyter output seems to capture some more details, as below.

Output for <code>Jupyter</code>:
</summary>

<p>

```
error 21:34:55.129: Execute_request failure enabling debugging for IW import debugpy;debugpy.listen(('localhost', 0)) RuntimeError: timed out waiting for adapter to connect
     [0;31m---------------------------------------------------------------------------[0m
    [0;31mtimeout[0m                                   Traceback (most recent call last)
    File [0;32m~/miniconda3/envs/python39_tf/lib/python3.9/site-packages/debugpy/server/api.py:218[0m, in [0;36mlisten[0;34m(address, settrace_kwargs)[0m
[1;32m    217[0m [38;5;28;01mtry[39;00m:
[0;32m--> 218[0m     sock, _ [38;5;241m=[39m [43mendpoints_listener[49m[38;5;241;43m.[39;49m[43maccept[49m[43m([49m[43m)[49m
[1;32m    219[0m     [38;5;28;01mtry[39;00m:

    File [0;32m~/miniconda3/envs/python39_tf/lib/python3.9/socket.py:293[0m, in [0;36msocket.accept[0;34m(self)[0m
[1;32m    287[0m [38;5;124;03m"""accept() -> (socket object, address info)[39;00m
[1;32m    288[0m 
[1;32m    289[0m [38;5;124;03mWait for an incoming connection.  Return a new socket[39;00m
[1;32m    290[0m [38;5;124;03mrepresenting the connection, and the address of the client.[39;00m
[1;32m    291[0m [38;5;124;03mFor IP sockets, the address info is a pair (hostaddr, port).[39;00m
[1;32m    292[0m [38;5;124;03m"""[39;00m
[0;32m--> 293[0m fd, addr [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_accept[49m[43m([49m[43m)[49m
[1;32m    294[0m sock [38;5;241m=[39m socket([38;5;28mself[39m[38;5;241m.[39mfamily, [38;5;28mself[39m[38;5;241m.[39mtype, [38;5;28mself[39m[38;5;241m.[39mproto, fileno[38;5;241m=[39mfd)

    [0;31mtimeout[0m: timed out
    
During handling of the above exception, another exception occurred:

    [0;31mRuntimeError[0m                              Traceback (most recent call last)
    Cell [0;32mIn[1], line 1[0m
[0;32m----> 1[0m [38;5;28;01mimport[39;00m [38;5;21;01mdebugpy[39;00m;[43mdebugpy[49m[38;5;241;43m.[39;49m[43mlisten[49m[43m([49m[43m([49m[38;5;124;43m'[39;49m[38;5;124;43mlocalhost[39;49m[38;5;124;43m'[39;49m[43m,[49m[43m [49m[38;5;241;43m0[39;49m[43m)[49m[43m)[49m

    File [0;32m~/miniconda3/envs/python39_tf/lib/python3.9/site-packages/debugpy/__init__.py:113[0m, in [0;36mlisten[0;34m(address)[0m
[1;32m     94[0m [38;5;124;03m"""Starts a debug adapter debugging this process, that listens for[39;00m
[1;32m     95[0m [38;5;124;03mincoming socket connections from clients on the specified address.[39;00m
[1;32m     96[0m 
[0;32m   (...)[0m
[1;32m    108[0m [38;5;124;03muntil the client connects.[39;00m
[1;32m    109[0m [38;5;124;03m"""[39;00m
[1;32m    111[0m [38;5;28;01mfrom[39;00m [38;5;21;01mdebugpy[39;00m[38;5;21;01m.[39;00m[38;5;21;01mserver[39;00m [38;5;28;01mimport[39;00m api
[0;32m--> 113[0m [38;5;28;01mreturn[39;00m [43mapi[49m[38;5;241;43m.[39;49m[43mlisten[49m[43m([49m[43maddress[49m[43m)[49m

    File [0;32m~/miniconda3/envs/python39_tf/lib/python3.9/site-packages/debugpy/server/api.py:143[0m, in [0;36m_starts_debugging.<locals>.debug[0;34m(address, **kwargs)[0m
[1;32m    141[0m     [38;5;28;01mreturn[39;00m func(address, settrace_kwargs, [38;5;241m*[39m[38;5;241m*[39mkwargs)
[1;32m    142[0m [38;5;28;01mexcept[39;00m [38;5;167;01mException[39;00m:
[0;32m--> 143[0m     [43mlog[49m[38;5;241;43m.[39;49m[43mreraise_exception[49m[43m([49m[38;5;124;43m"[39;49m[38;5;132;43;01m{0}[39;49;00m[38;5;124;43m() failed:[39;49m[38;5;124;43m"[39;49m[43m,[49m[43m [49m[43mfunc[49m[38;5;241;43m.[39;49m[38;5;18;43m__name__[39;49m[43m,[49m[43m [49m[43mlevel[49m[38;5;241;43m=[39;49m[38;5;124;43m"[39;49m[38;5;124;43minfo[39;49m[38;5;124;43m"[39;49m[43m)[49m

    File [0;32m~/miniconda3/envs/python39_tf/lib/python3.9/site-packages/debugpy/server/api.py:141[0m, in [0;36m_starts_debugging.<locals>.debug[0;34m(address, **kwargs)[0m
[1;32m    138[0m settrace_kwargs[[38;5;124m"[39m[38;5;124mdont_trace_end_patterns[39m[38;5;124m"[39m] [38;5;241m=[39m ([38;5;28mstr[39m([38;5;124m"[39m[38;5;124mdebugpy_launcher.py[39m[38;5;124m"[39m),)
[1;32m    140[0m [38;5;28;01mtry[39;00m:
[0;32m--> 141[0m     [38;5;28;01mreturn[39;00m [43mfunc[49m[43m([49m[43maddress[49m[43m,[49m[43m [49m[43msettrace_kwargs[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m)[49m
[1;32m    142[0m [38;5;28;01mexcept[39;00m [38;5;167;01mException[39;00m:
[1;32m    143[0m     log[38;5;241m.[39mreraise_exception([38;5;124m"[39m[38;5;132;01m{0}[39;00m[38;5;124m() failed:[39m[38;5;124m"[39m, func[38;5;241m.[39m[38;5;18m__name__[39m, level[38;5;241m=[39m[38;5;124m"[39m[38;5;124minfo[39m[38;5;124m"[39m)

    File [0;32m~/miniconda3/envs/python39_tf/lib/python3.9/site-packages/debugpy/server/api.py:232[0m, in [0;36mlisten[0;34m(address, settrace_kwargs)[0m
[1;32m    228[0m [38;5;28;01mexcept[39;00m socket[38;5;241m.[39mtimeout:
[1;32m    229[0m     log[38;5;241m.[39mswallow_exception(
[1;32m    230[0m         [38;5;124m"[39m[38;5;124mTimed out waiting for adapter to connect:[39m[38;5;124m"[39m, level[38;5;241m=[39m[38;5;124m"[39m[38;5;124minfo[39m[38;5;124m"[39m
[1;32m    231[0m     )
[0;32m--> 232[0m     [38;5;28;01mraise[39;00m [38;5;167;01mRuntimeError[39;00m([38;5;124m"[39m[38;5;124mtimed out waiting for adapter to connect[39m[38;5;124m"[39m)
[1;32m    233[0m [38;5;28;01mexcept[39;00m [38;5;167;01mException[39;00m [38;5;28;01mas[39;00m exc:
[1;32m    234[0m     log[38;5;241m.[39mswallow_exception([38;5;124m"[39m[38;5;124mError retrieving adapter endpoints:[39m[38;5;124m"[39m, level[38;5;241m=[39m[38;5;124m"[39m[38;5;124minfo[39m[38;5;124m"[39m)

    [0;31mRuntimeError[0m: timed out waiting for adapter to connect
error 21:34:55.129: Error in execution (get message for cell) [o [Error]: RuntimeError
	at b.connectToLocal (/Users/tanay/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/out/extension.node.js:2:1605659)
	at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async b.connect (/Users/tanay/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/out/extension.node.js:2:1604020)
	at async b.startDebugSession (/Users/tanay/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/out/extension.node.js:2:1603163)
	at async z.createExecutionPromise (/Users/tanay/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/out/extension.node.js:2:1695955)] {
  category: 'notinstalled',
  kernelConnectionMetadata: f {
    kind: 'startUsingPythonInterpreter',
    kernelSpec: {
      specFile: '/Users/tanay/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/temp/jupyter/kernels/pythonjvsc74a57bd09e459d461750e0ad0dd48f0512403b45939cf5739455aca8620f655ec5e30e13/kernel.json',
      interpreterPath: '/Users/tanay/miniconda3/envs/python39_tf/bin/python',
      isRegisteredByVSC: 'registeredByNewVersionOfExt',
      name: 'pythonjvsc74a57bd09e459d461750e0ad0dd48f0512403b45939cf5739455aca8620f655ec5e30e13',
      argv: [Array],
      language: 'python',
      executable: 'python',
      display_name: 'python39_tf',
      metadata: [Object],
      env: {},
      interrupt_mode: undefined
    },
    interpreter: {
      id: '/Users/tanay/miniconda3/envs/python39_tf/bin/python',
      sysPrefix: '/Users/tanay/miniconda3/envs/python39_tf',
      envPath: [d],
      displayPath: [d],
      envName: 'python39_tf',
      uri: [d],
      displayName: 'python39_tf',
      envType: 'Conda',
      version: [Object]
    },
    id: '.jvsc74a57bd09e459d461750e0ad0dd48f0512403b45939cf5739455aca8620f655ec5e30e13./Users/tanay/miniconda3/envs/python39_tf/python./Users/tanay/miniconda3/envs/python39_tf/python.-m#ipykernel_launcher'
  }
}
warn 21:34:55.130: DataScience Error [o [Error]: RuntimeError
	at b.connectToLocal (/Users/tanay/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/out/extension.node.js:2:1605659)
	at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async b.connect (/Users/tanay/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/out/extension.node.js:2:1604020)
	at async b.startDebugSession (/Users/tanay/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/out/extension.node.js:2:1603163)
	at async z.createExecutionPromise (/Users/tanay/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/out/extension.node.js:2:1695955)] {
  category: 'notinstalled',
  kernelConnectionMetadata: f {
    kind: 'startUsingPythonInterpreter',
    kernelSpec: {
      specFile: '/Users/tanay/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/temp/jupyter/kernels/pythonjvsc74a57bd09e459d461750e0ad0dd48f0512403b45939cf5739455aca8620f655ec5e30e13/kernel.json',
      interpreterPath: '/Users/tanay/miniconda3/envs/python39_tf/bin/python',
      isRegisteredByVSC: 'registeredByNewVersionOfExt',
      name: 'pythonjvsc74a57bd09e459d461750e0ad0dd48f0512403b45939cf5739455aca8620f655ec5e30e13',
      argv: [Array],
      language: 'python',
      executable: 'python',
      display_name: 'python39_tf',
      metadata: [Object],
      env: {},
      interrupt_mode: undefined
    },
    interpreter: {
      id: '/Users/tanay/miniconda3/envs/python39_tf/bin/python',
      sysPrefix: '/Users/tanay/miniconda3/envs/python39_tf',
      envPath: [d],
      displayPath: [d],
      envName: 'python39_tf',
      uri: [d],
      displayName: 'python39_tf',
      envType: 'Conda',
      version: [Object]
    },
    id: '.jvsc74a57bd09e459d461750e0ad0dd48f0512403b45939cf5739455aca8620f655ec5e30e13./Users/tanay/miniconda3/envs/python39_tf/python./Users/tanay/miniconda3/envs/python39_tf/python.-m#ipykernel_launcher'
  }
}
```

</p>
</details>

<details>

<summary>User Settings</summary>

<p>

```

languageServer: "Pylance"

```

</p>
</details>

Extension version: 2022.20.2
VS Code version: Code 1.74.3 (Universal) (97dec172d3256f8ca4bfb2143f3f76b503ca0534, 2023-01-09T17:07:18.579Z)
OS version: Darwin arm64 22.2.0
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Pro (8 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|16, 11, 8|
|Memory (System)|32.00GB (0.06GB free)|
|Process Argv|--crash-reporter-id 1dd3667b-8e78-4216-9bf8-a017dcfcee65|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30445986
pythondataviewer:30285071
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30410667
cppdebug:30492333
vscaac:30438847
vsclangdf:30486550
c4g48928:30535728
dsvsc012:30540252
azure-dev_surveyone:30548225
vscccc:30610679
pyindex848cf:30577861
nodejswelcome1:30587005
3biah626:30602489
iaj6b796:30613358
f6dab269:30613381
fim-prod:30623723

```

</details>

**Comment by roblourens** [comment]
The initial error is inside debugpy, I'm not sure how to interpet it, I will move it there for the first look. Cleaned up the initial error a bit:

```
error 21:34:55.129: Execute_request failure enabling debugging for IW import debugpy;debugpy.listen(('localhost', 0)) RuntimeError: timed out waiting for adapter to connect
     ---------------------------------------------------------------------------
    timeout                                   Traceback (most recent call last)
    File ~/miniconda3/envs/python39_tf/lib/python3.9/site-packages/debugpy/server/api.py:218, in listen(address, settrace_kwargs)
try:
--> 218     sock, _ = �endpoints_listener.accept()
    try:

    File ~/miniconda3/envs/python39_tf/lib/python3.9/socket.py:293, in socket.accept(self)
"""accept() -> (socket object, address info)

Wait for an incoming connection.  Return a new socket
representing the connection, and the address of the client.
For IP sockets, the address info is a pair (hostaddr, port).
"""
--> 293 fd, addr = self._accept()
sock = socket(elf.family, elf.type, elf.proto, fileno=fd)

    timeout: timed out
```

**Comment by roblourens** [comment]
@ttopac can you check your debugpy version?

```
import debugpy
debugpy.__version__
```

in that kernel.

Will leave it here for now as I don't have permission to it to debugpy anyway, @int19h can you take a look?

**Comment by ttopac** [comment]
Sorry, but I had to make some changes in the environment (package installations that came with some updates to the existing packages as well), and the issue got resolved itself. Possibly, one package had a conflict. Sorry that I won't be able to provide any more debugging info.

**Comment by DonJayamanne** [comment]
> Sorry that I won't be able to provide any more debugging info.

Thank you for gettingb back.
Closing this as stale
